src/components/shared/video-card/VideoCardActionMenu.jsx

```diff
@@ -3,66 +3,73 @@ import { MdThumbDown, MdThumbUp, MdVideoLibrary } from "react-icons/md";
 import { RiPlayList2Line, RiShareForwardLine } from "react-icons/ri";
 import { useLikedVideos } from "../../../context/liked-videos/LikedVideoContext";
 import { usePlaylist } from "../../../context/playlist/PlaylistContext";
+import { useWatchLaterVideos } from "../../../context/watch-later/WatchLaterVideoContext";
 import PlaylistModal from "../../playlist/PlaylistModal";
 import DropdownMenu from "../DropdownMenu";
 import Modal from "../modal/Modal";
 
-const VideoCardActionMenu = ({ videoId, playlistId, isInLiked }) => {
-  const { addToLikes, removeFromLikes } = useLikedVideos();
+const VideoCardActionMenu = ({ videoId, playlistId }) => {
+  const { addToLikes, removeFromLikes, likedVideos } = useLikedVideos();
   const [showModal, setShowModal] = useState(false);
   const { removeFromPlaylist } = usePlaylist();
+  const { watchLaterVideos, addToWatchLater, removeFromWatchLater } =
+    useWatchLaterVideos();
 
-  const addToPlayList = (id) => {
+  const isLiked = likedVideos.some((video) => video._id === videoId);
+
+  const isInWatchLater = watchLaterVideos.some(
+    (video) => video._id === videoId
+  );
+
+  const addToPlayListClickHandler = () => {
     if (!playlistId) setShowModal((v) => !v);
     else removeVideoHandler(videoId, playlistId);
   };
 
-  const likeDislikeClickHandler = () => {
-    if (isInLiked) removeFromLikes(videoId);
-    else addToLikes(videoId);
-  };
+  const likeDislikeClickHandler = () =>
+    isLiked ? removeFromLikes(videoId) : addToLikes(videoId);
 
-  const removeVideoHandler = (videoId, playlistId) => {
-    if (videoId || playlistId) removeFromPlaylist({ playlistId, videoId });
-  };
+  const removeVideoHandler = (videoId, playlistId) =>
+    playlistId && removeFromPlaylist({ playlistId, videoId });
+
+  const watchLaterClickHandler = () =>
+    isInWatchLater ? removeFromWatchLater(videoId) : addToWatchLater(videoId);
```

Comment:
defining separate functions for repeating task is an good idea, liked it.

Reply:
thanks